sympy/functions/elementary/trigonometric.py

```diff
@@ -2251,6 +2251,16 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_is_real(self):
+        x = self.args[0]
+        if x.is_real == False:
+            return False
+        y = abs(x) - 1
```

Comment:
Also the result is true strictly speaking. The truth value cannot, in fact, be determined for a symbol x without further restrictions. So it need not have anything to do with asec(0) not being defined as I first suspected. (Though 'type error' may not be the best way of returning the outcome.)
One should give enough data for assumptions to be able to determine the truth value.